src/main/compa/controllers/FriendshipController.java

```diff
@@ -1,143 +1,125 @@
 package main.compa.controllers;
 
-import com.google.gson.GsonBuilder;
-import com.mongodb.BasicDBObject;
-import com.mongodb.DBCollection;
-import com.mongodb.DBObject;
-import com.mongodb.DBCursor;
-import main.compa.app.*;
-import com.google.gson.Gson;
+import io.vertx.core.json.JsonArray;
+import main.compa.app.Container;
 import main.compa.daos.FriendshipDAO;
 import main.compa.daos.UserDAO;
 import main.compa.dtos.UserDTO;
-import main.compa.exception.FriendshipException;
 import io.vertx.core.http.HttpMethod;
-import io.vertx.ext.web.Router;
 import io.vertx.ext.web.RoutingContext;
 import main.compa.app.Controller;
-import main.compa.app.ModelManager;
-import main.compa.app.ServiceManager;
-import main.compa.daos.FriendshipDAO;
-import main.compa.daos.UserDAO;
-import main.compa.exception.FriendshipException;
-import main.compa.exception.RegisterException;
 import main.compa.models.Friendship;
 import main.compa.models.User;
-import main.compa.models.Friendship;
-import main.compa.models.User;
-import org.mongodb.morphia.query.UpdateOperations;
+import main.compa.services.GsonService;
 
-import javax.management.Query;
 import java.util.List;
 
 public class FriendshipController extends Controller {
-    private static final String PREFIX = "/friend";
+    private static final String PREFIX = "/friendship";
 
     private FriendshipDAO friendshipDAO;
     private UserDAO userDAO;
 
-    Gson gson = new GsonBuilder().setPrettyPrinting().create();
-    //TODO CHANGE AUTH TOKEN TO THE HEADER AND NOT THE BODY
-
-    public FriendshipController(ServiceManager serviceManager, Router router, ModelManager modelManager) {
-        super(serviceManager, PREFIX, router);
-        this.registerRoute(HttpMethod.POST, "/", this::addFriendship, "application/json");
-        this.registerRoute(HttpMethod.GET, "/getFriend", this::getFriends, "application/json");
+    public FriendshipController(Container container) {
+        super(PREFIX, container);
+        this.registerAuthRoute(HttpMethod.POST, "/request", this::requestFriendship, "application/json");
+        this.registerAuthRoute(HttpMethod.POST, "/response", this::respondToRequest, "application/json");
+        this.registerAuthRoute(HttpMethod.GET, "/friend_list", this::getFriends, "application/json");
+        this.registerAuthRoute(HttpMethod.GET, "/pending", this::getPendingFriendships, "application/json");
 
-        friendshipDAO = (FriendshipDAO) modelManager.getDAO(Friendship.class);
-        userDAO = (UserDAO) modelManager.getDAO(User.class);
+        friendshipDAO = (FriendshipDAO) container.getDAO(Friendship.class);
```

Comment:
@ssssarah we can access to the container from a controller subclass ?  It sounds dangerous ☄️ Container should be private inside the Controller superclass, no ?

Reply:
it is, but in order to call the super method, we need the container as a parameter of the constructor. Therefore, the container can be accessed inside the constructor, but not outside (in any other method of the class)